src/core/public/execution_context/execution_context_service.ts

```diff
@@ -134,4 +141,45 @@ export class ExecutionContextService
       ...context,
     };
   }
+
+  /**
+   * Sets the analytics context provider based on the execution context details.
+   * @param analytics The analytics service
+   * @private
+   */
+  private enrichAnalyticsContext(analytics: AnalyticsServiceSetup) {
+    analytics.registerContextProvider({
+      name: 'execution_context',
```

Comment:
FWIW, the `name` in the context provider API ID only used to identify it in case we want to unregister it. It doesn't affect the output.

I was trying to follow
- if it comes out of a service, provide the service `execution_context`
- otherwise, use a descriptive name

Re conventions, the @elastic/platform-analytics  team are looking into creating a guide with preferred naming